### ISSUE TITLE
TypeError during an upload_session with the chunked_uploader

### ISSUE THREAD

**Issue by anpr** [box/box-python-sdk#483]
- [x] I have checked that the [SDK documentation][sdk-docs] and [API documentation][api-docs] doesn't solve my issue

### Description of the Issue
In our code, with the update of box SDK to 2.7.0 our code that uploads file with the chunked_uploader breaks, with the following exception:
`TypeError: a bytes-like object is required, not 'str'`

### Versions Used
Box Python SDK: 2.7.0
Python: 3.7.3

### Steps to Reproduce
Our code triggers the chunked upload like this:
```
chunked_uploader = BoxClient.get_chunked_uploader_for_folder(
    folder=self.folder(folder_id), file_path=file_path, file_name=file_name
)
return chunked_uploader.start()
```

### Error Message, Including Traceback
Stacktrace, up to code above (not included):
```
 /usr/local/lib/python3.7/site-packages/boxsdk/util/chunked_uploader.py(51)start()
     49         if self._is_aborted:
     50             raise BoxException('The upload has been previously aborted. Please retry upload with a new upload session.')
---> 51         self._upload()
     52         content_sha1 = self._sha1.digest()
     53         return self._upload_session.commit(content_sha1=content_sha1, parts=self._part_array)

  /usr/local/lib/python3.7/site-packages/boxsdk/util/chunked_uploader.py(107)_upload()
    105             self._sha1.update(next_part.chunk)
    106             # Retrieve the uploaded part if the part has already been uploaded. If not upload the current part.
--> 107             uploaded_part = self._part_definitions.get(next_part.offset) or next_part.upload()
    108             self._inflight_part = None
    109             # Record that the part has been uploaded.

  /usr/local/lib/python3.7/site-packages/boxsdk/util/chunked_uploader.py(193)upload()
    190         return self._upload_session.upload_part_bytes(
    191             part_bytes=self.chunk,
    192             offset=self.offset,
--> 193             total_size=self._total_size
    194         )

  /usr/local/lib/python3.7/site-packages/boxsdk/object/upload_session.py(98)upload_part_bytes()
     96             self.get_url(),
     97             headers=headers,
---> 98             data=part_bytes,
     99         )
    100         return response.json()['part']

  /usr/local/lib/python3.7/site-packages/boxsdk/session/session.py(122)put()
    120             `unicode`
    121         """
--> 122         return self.request('PUT', url, **kwargs)
    123 
    124     def delete(self, url, **kwargs):

  /usr/local/lib/python3.7/site-packages/boxsdk/session/session.py(158)request()
    156             `unicode`
    157         """
--> 158         response = self._prepare_and_send_request(method, url, **kwargs)
    159         return self.box_response_constructor(response)
    160 

  /usr/local/lib/python3.7/site-packages/boxsdk/session/session.py(397)_prepare_and_send_request()
    395 
    396         while True:
--> 397             retry = self._get_retry_request_callable(network_response, attempt_number, request, **kwargs)
    398 
    399             if retry is None or attempt_number >= API.MAX_RETRY_ATTEMPTS:

  /usr/local/lib/python3.7/site-packages/boxsdk/session/session.py(594)_get_retry_request_callable()
    592             attempt_number,
    593             request,
--> 594             **kwargs
    595         )
    596 

  /usr/local/lib/python3.7/site-packages/boxsdk/session/session.py(440)_get_retry_request_callable()
    438         data = kwargs.get('data', {})
    439         grant_type = None
--> 440         if 'grant_type' in data:
    441             grant_type = data['grant_type']
    442         code = network_response.status_code
```

Error Message:
`TypeError: a bytes-like object is required, not 'str'`

I suspect the last line of the stack trace, which has been introduced with 2.7.0 is the culprit. `grant_type` is a `str`, whereas `data` is a bytes-object.

**Comment by anpr** [comment]
Box SDK 2.7.0, not 3.7.0, sorry.

**Comment by PJSimon** [comment]
Hi @anpr and @samkuehn!  We're taking a look right now.  Thanks!

**Comment by PJSimon** [comment]
Hi @anpr and @samkuehn,
Thanks for reporting this issue, and thanks for your patience.  This has been resolved in the latest release, here: https://github.com/box/box-python-sdk/releases/tag/v2.7.1

Please let us know if you experience any further issues.  Thanks!

**Comment by dominik-bln** [comment]
Thanks for the quick fix, v2.7.1 looks good for us.

**Comment by anpr** [comment]
Yes, thanks for the fast reaction!
